### PR TITLE
Refactored the CfpController to enable this class to be tested by unittests in the near future

### PR DESCRIPTION
**Pull request by staal-it**
Refactored the CfpController to enable this class to be tested by unittests in the near future. Please let me know if this direction works for you.

### CODE REVIEW

File: CfpExchange/Services/CfpService.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            return _cfpContext.Cfps.Any(cfp => cfp.Slug == cfpToAddSlug);
+        }
+
+        public void AddCfp(Cfp cfpToAdd)
+        {
+            _cfpContext.Add(cfpToAdd);
+        }
+
+        public void SaveChanges()
```
Comment (jfversluis) [inline CfpExchange/Services/CfpService.cs]: What is the benefit of exposing these separately? Why not just do this whenever you do a create or update?

Comment (staal-it) [reply CfpExchange/Services/CfpService.cs]: I guess you commented on the two SaveChanges[Async] methods? They were already being using both, so I simply added them both to the new service. I removed the non-async SaveChanges.

File: CfpExchange/Controllers/CfpController.cs [hunk 2]
```diff
@@ -17,379 +13,341 @@
...
+            try
+            {
+                var linkedCfp = _cfpService.GetCfpById(id);
+
+                if (linkedCfp != null)
+                {
+                    linkedCfp.ClicksToCfpUrl++;
+
+                    await _cfpService.SaveChangesAsync();
+                }
```
Comment (staal-it) [inline CfpExchange/Controllers/CfpController.cs]: Now the .Single is gone and there's a check for null I think we could remove the try ... catch right?

Comment (jfversluis) [reply CfpExchange/Controllers/CfpController.cs]: Yep! I don't see a reason why not!

Comment (staal-it) [reply CfpExchange/Controllers/CfpController.cs]: Removed

**Pull request merged**